pkg/pillar/types/zedroutertypes.go

```diff
@@ -3535,3 +3544,40 @@ type WwanSignalInfo struct {
 	RSRP int32 `json:"rsrp"`
 	SNR  int32 `json:"snr"`
 }
+
+// WwanLocationInfo contains device location information obtained from a GNSS
+// receiver integrated into an LTE modem.
+type WwanLocationInfo struct {
+	// Latitude in the Decimal degrees (DD) notation.
+	Latitude float64 `json:"latitude"`
+	// Longitude in the Decimal degrees (DD) notation.
+	Longitude float64 `json:"longitude"`
+	// Altitude w.r.t. mean sea level in meters.
+	Altitude float64 `json:"altitude"`
+	// Circular horizontal position uncertainty in meters.
+	HorizontalUncertainty float32 `json:"horizontal-uncertainty"`
+	// Reliability of the provided information for latitude and longitude.
+	HorizontalReliability LocReliability `json:"horizontal-reliability"`
+	// Vertical position uncertainty in meters.
+	VerticalUncertainty float32 `json:"vertical-uncertainty"`
+	// Reliability of the provided information for altitude.
+	VerticalReliability LocReliability `json:"vertical-reliability"`
+	// Unix timestamp in milliseconds.
+	UTCTimestamp uint64 `json:"utc-timestamp"`
```

Comment:
Is this limited to milliseconds by the GNSS? Or the way the modem reports it?

Reply:
I think this is limited by how the modem reports it and the fact that it is integrated with an LTE modem.
With standalone GNSS modules there is a separate serial connection for PPS service (https://www.kernel.org/doc/html/latest/driver-api/pps.html), allowing high-precision time synchronization.
With GNSS receiver integrated into LTE modem we only get a single serial interface streaming NMEA traces. This can be used to determine location but not to synchronize clocks with high precision.